db/memory.go

```diff
@@ -43,6 +43,10 @@ func (r *memoryDatabase) StorePreset(preset types.Preset) (types.Preset, error)
 	r.mtx.Lock()
 	defer r.mtx.Unlock()
 
+	if _, ok := r.presets[preset.Name]; ok {
```

Comment:
I wanted to call RetrivePreset here too but the locks stopped me. Didn't come up with a way to resolve it so went this way for now.